docs/features/nullable-reference-types.md

```diff
@@ -0,0 +1,119 @@
+Nullable Reference Types
+=========
+References to reference types may be nullable, non-nullable, or null-oblivious. (The document uses the shorthand `?`, `!`, and `~`.)
+
+## Annotations
+In source, nullable reference types are annotated with `?`.
+```c#
+string? OptString; // may be null
+List<object?>? OptListOptElements; // list may be null, elements may be null
+```
+In metadata, nullable reference types are annotated with a `[Nullable]` attribute.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    public sealed class NullableAttribute : Attribute
+    {
+        public NullableAttribute() { }
```

Comment:
This empty constructor is there to fulfill some constraint, but will not be used or be useful. Correct?

Reply:
The parameter-less constructor is used for a nullable reference type with no nested types: `string? MaybeNull;`

---
In reply to: [194861141](https://github.com/dotnet/roslyn/pull/27721#discussion_r194861141) [](ancestors = 194861141)